language/vm/src/file_format.rs

```diff
@@ -374,17 +375,36 @@ pub struct FieldDefinition {
     pub signature: TypeSignature,
 }
 
+/// `VisibilityScope` restricts who may call a given function.
+#[derive(Clone, Debug, Eq, PartialEq)]
+#[cfg_attr(any(test, feature = "fuzzing"), derive(Arbitrary))]
+#[cfg_attr(any(test, feature = "fuzzing"), proptest(no_params))]
+pub enum VisibilityScope {
+    /// The function can only be called within the module that defines it.
+    Private,
+    /// The function can be called by any module and script.
+    Public,
+    /// Similar to "private", but in addition, the function can also be called
+    /// by any module specified in the list of module handles.
+    Protected(Vec<ModuleHandleIndex>),
```

Comment:
I'd probably break this out to 

`Protected(VisibilityListIndex)`

and then somewhere else
```
define_index! {
    name: VisibilityListIndex,
    kind: VisibilityList,
    doc: "Index into the `VisibilityList` table.",
}
struct VisibilityList {
    modules: Vec<ModuleHandleIndex>
}
```

I imagine a lot of functions will have the same set of valid callers. 
It will also reduce the diff size on updates to the list